### PR TITLE
PROD-2116 - Hidden conversations count remove from the total message count. 

### PR DESCRIPTION
**Pull request by krunal-buddyboss**
### Jira Issue:
https://buddyboss.atlassian.net/browse/PROD-2116

### General Note
Keep all conversations related to this PR in the associated Jira issue(s). Do NOT add comment on this PR or edit this PR’s description.

### Notes to Developer
* Ensure the IDs (i.e. PROD-1) of all associated Jira issues are reference in this PR’s title
* Ensure that you have achieved the Definition of Done before submitting for review
* When this PR is ready for review, move the associate Jira issue(s) to “Needs Review” (or “Code Review” for Dev Tasks)

### Notes to Reviewer
* Ensure that the Definition of Done have been achieved before approving a PR
* When this PR is approved, move the associated Jira issue(s) to “Needs QA” (or “Approved” for Dev Tasks)

### CODE REVIEW

File: src/bp-templates/bp-nouveau/includes/messages/ajax.php [hunk 1]
```diff
@@ -1082,7 +1082,7 @@ function bp_nouveau_ajax_get_user_message_threads() {
 			'group_message_fresh'             => $group_message_fresh,
 			'excerpt'                         => wp_strip_all_tags( bp_get_message_thread_excerpt() ),
 			'content'                         => do_shortcode( bp_get_message_thread_content() ),
-			'unread'                          => bp_message_thread_has_unread(),
+			'unread'                          => bb_get_messages_thread_unread_count( $bp_get_message_thread_id ) ? true : false,
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/messages/ajax.php:1085]: @krunal-buddyboss We don't need to change anything here

Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/messages/ajax.php:1085]:
```suggestion
			'unread'                          => bp_message_thread_has_unread(),
```
@krunal-buddyboss We don't want to change anything here revert this change.

File: src/bp-messages/bp-messages-functions.php [hunk 2]
```diff
@@ -622,6 +622,38 @@ function messages_get_unread_count( $user_id = 0 ) {
...
+		return;
+	}
+
+	$cache_key = 'bb_message_thread_unread_count_' . $user_id . '_' . $thread_id;
+	$unread_count = wp_cache_get( $cache_key, 'bp_messages_unread_count' );
+
+	if ( false === $unread_count ) {
+		global $wpdb;
+		$bp = buddypress();
+		$unread_count = (int) $wpdb->get_var( $wpdb->prepare( "SELECT unread_count FROM {$bp->messages->table_name_recipients} WHERE user_id = %d AND thread_id = %d and is_deleted = 0 AND sender_only = 0", $user_id, $thread_id ) );
```
Comment (chetansatasiya) [inline src/bp-messages/bp-messages-functions.php:651]:
```suggestion
		$unread_count = (int) $wpdb->get_col( $wpdb->prepare( "SELECT unread_count from {$bp->messages->table_name_recipients} WHERE user_id = %d AND thread_id = %d AND unread_count > 0", $user_id, $thread_id ) );
```

File: src/bp-messages/bp-messages-functions.php [hunk 3]
```diff
@@ -622,6 +622,38 @@ function messages_get_unread_count( $user_id = 0 ) {
...
+ * Get the thread unread messages count for a user.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param int $thread_id Thread ID of the message.
+ * @param int $user_id Optional. ID of the user. Default: ID of the logged-in user.
+ *
+ * @return int
+ */
+function bb_get_messages_thread_unread_count( $thread_id, $user_id = 0 ) {
```
Comment (chetansatasiya) [inline src/bp-messages/bp-messages-functions.php:635]:
```suggestion
function bb_get_thread_messages_unread_count( $thread_id, $user_id = 0 ) {
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@krunal-buddyboss We need to change the condition here https://github.com/buddyboss/buddyboss-platform/blob/PROD-2116/src/bp-messages/classes/class-bp-messages-thread.php#L206#L208 and use the function here to get unread count.

File: src/bp-templates/bp-nouveau/includes/messages/ajax.php (hunk 1 above)
Comment (krunal-buddyboss) [reply src/bp-templates/bp-nouveau/includes/messages/ajax.php]: Done, Applied changes.

File: src/bp-messages/bp-messages-functions.php (hunk 2 above)
Comment (krunal-buddyboss) [reply src/bp-messages/bp-messages-functions.php:651]: Done, Applied changes.

File: src/bp-messages/bp-messages-functions.php (hunk 3 above)
Comment (krunal-buddyboss) [reply src/bp-messages/bp-messages-functions.php:635]: Done, Applied changes.

**Review by KartikSuthar — DISMISSED** [review]
